### DISCUSSION TITLE
Limit on the number of smart meters?

### DISCUSSION THREAD

**Discussion by esplesekopf** [arendst/Tasmota#15689] [Q&A] [answered]
Hello.

Is there a limit in the number of smart meters that can be processed to 3 pieces?

I have an ESP32 to which I have connected 4 of my IR readers. 
Each reader individually works. But together only the first 3 work.

I can swap the GPIO definitions in the script for meters 1-4 arbitrarily, always meter 4 does not generate any data.

Thank you very much!

**Comment by sfromis** [comment]
While I'm not sure of details of smart meter interface, the general Tasmota energy monitoring setup is limited to 3 sets of power/energy values, 3 phases. OTOH, docs does mention `MAX_METERS` having a default of 5.

**Comment by gemu2015** [comment, marked as answer]
esp32 has 3 usable hardware serial lines, as such limiting the number of serial meters to 3.

however there is an additional esp32 software serial option that may be enabled by
#define USE_ESP32_SW_SERIAL
a negative receive pin number denotes to use software serial, ESP32 software only supports receive no transmit

this allows for more more than 3 serial Meters.

keep in mind that there also may be other limitations described in the docs that may be overridden by certain #defines like SML_MAX_VARS

**Reply by esplesekopf** [reply]
Thank you!

**Comment by esplesekopf** [comment]
Super information!

I will try it with the software Serial and get back to you. 

That the software Serial can only receive is sufficient for me. Basically 90% of the meters out there are PushData meters. If there is one in an installation that needs to be addressed, I have another 3 channels.

My test equipment includes an ISKRA, an EasyMeter, an ISKRA Simulator (all PushData) and a Siemens TD3511. This must be read out with start sequence and baud rate switching. If this combination works together, I am satisfied. At the moment this combination works, but only 3 of the meters.

I'll go compile Tasmota:)

**Comment by esplesekopf** [comment]
Perfect!

With the software Serial Option I can now read all 4 meters. Per MQTT also all values arrive.  

Thanks for the help!
